pkg/simulator/terraform_vars_test.go

```diff
@@ -15,15 +17,19 @@ access_cidr = "10.0.0.1/16"
 attack_container_tag = "latest"
 state_bucket_name = "test-bucket"
 `
-	assert.Equal(t, tfv.String(), expected)
+	assert.Equal(t, expected, tfv.String())
 }
 
 func Test_Ensure_TfVarsFile_with_settings(t *testing.T) {
 	tfDir := fixture("tf-dir-with-settings")
 	varsFile := tfDir + "/settings/bastion.tfVars"
 
 	err := simulator.EnsureLatestTfVarsFile(tfDir, "ssh-rsa", "10.0.0.1/16", "test-bucket", "latest")
-	assert.Nil(t, err, "Got an error")
-
-	assert.Equal(t, util.MustSlurp(varsFile), "test = true\n")
+	require.NoError(t, err)
+	expected := `access_key = "ssh-rsa"
```

Comment:
I think this should be as simple as `expected := "test = test\n". The rest of these vars and their values do not appear when I run `make run` on my Ubuntu machine.

Reply:
Error output:
```
--- FAIL: Test_Ensure_TfVarsFile_with_settings (0.01s)
    terraform_vars_test.go:34: 
                Error Trace:    terraform_vars_test.go:34
                Error:          Not equal: 
                                expected: "access_key = \"ssh-rsa\"\naccess_cidr = \"10.0.0.1/16\"\nattack_container_tag = \"latest\"\nstate_bucket_name = \"test-bucket\"\n"
                                actual  : "test = true\n"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,5 +1,2 @@
                                -access_key = "ssh-rsa"
                                -access_cidr = "10.0.0.1/16"
                                -attack_container_tag = "latest"
                                -state_bucket_name = "test-bucket"
                                +test = true
                                 
                Test:           Test_Ensure_TfVarsFile_with_settings
FAIL
coverage: 56.7% of statements
FAIL    github.com/controlplaneio/simulator-standalone/pkg/simulator    7.397s
ok      github.com/controlplaneio/simulator-standalone/pkg/ssh  1.037s  coverage: 27.7% of statements
ok      github.com/controlplaneio/simulator-standalone/pkg/util 7.896s  coverage: 76.6% of statements
FAIL
make: *** [Makefile:147: test-unit] Error 1
The command '/bin/sh -c make test-unit' returned a non-zero code: 2
Makefile:95: recipe for target 'docker-build' failed
make: *** [docker-build] Error 2
```

Reply:
Just to confirm I got the same error message running on OS X.

Reply:
What is the expected behaviour? In the code I can see an [overwrite](https://github.com/kubernetes-simulator/simulator/blob/master/pkg/simulator/terraform_vars.go#L36) which should persist the new vars. 
The method doc + previous test assumption `test = true` are defining the opposite.

Reply:
The test before tests that the `TfVars` struct serialises to a string correctly when fully populated.  This test checks that when `EnsureTfVars` is run with a pre-existing settings file that it doesn't overwrite it. (Sorry I commented on this yesterday and somehow lost my review!)

Reply:
If you look here:
https://github.com/kubernetes-simulator/simulator/blob/7355b0cc3d65ce9f0ca00a151ca6a2645bb684a7/pkg/simulator/terraform_vars_test.go#L22-L23
We stub out the terraform directory with some dummy files.  The `bastion.tfVars` in the fixture simply contains the string "test = true\n" so if you revert the assertion as @s-irvine suggested this test will go green again.  I think we can merge at that point :)

Reply:
🤔 You expect the file [bastion.tfVars](https://github.com/kubernetes-simulator/simulator/blob/master/test/fixtures/tf-dir-with-settings/settings/bastion.tfVars) not to be overwritten but it is on my machine. Also I read the code that it should always overwrite the file.

Please note that the code was modified in [4c00e27831308d2d284269bd91c5f9f50184bd16](https://github.com/kubernetes-simulator/simulator/commit/4c00e27831308d2d284269bd91c5f9f50184bd16#diff-0cb36ecf8b239726eb4788f508d85cf6R32) from `util.EnsureFile` to `util.OverwriteFile`.
Neither doc nor tests were updated for that function to match the changed behaviour.

```sh
go test -v -count=1 ./pkg/simulator --run Test_Ensure_TfVarsFile_with_settings
```

Reply:
You are quite right :fearful: ... sorry it's been a while since I looked at this specific piece of code and as I remember it was written as we sprinted to get an initial release out to demo :cowboy_hat_face:! The **expected** behaviour is for that function to keep the tfvars up-to-date in case conditions change on the users machine.  I'm not sure why the assertion is that way or why its passing off the top of my head, Im afraid.  

Aside: You should be aware the large refactoring I mentioned has just been merged and this changes where the tfvars are written so this may affect this PR.  If I can help with specifics please do ask away and I'll do my best :smile: Sorry for the confusion!  The OSX fixes, terraform upgrade and assertion ordering fixes will all be super helpful

Reply:
👍 I have rebased on top of master, now

Reply:
Thanks :+1: Ive still got that failing test when I run `make docker-test` though.  I think we need to get to the bottom of that before I can merge.

Reply:
@raoulmillais can you try again please? I have upgraded the Go version in the Dockerfile. I was using a more recent one as well as with `golang:latest` container.

Reply:
thanks @alpe - upgrading go is great. evergreen ftw :) unfortunately the test is still failing both inside and outside the container (`make test` and `make docker-test`)

Reply:
ok, finally fixed thanks to @husio . OSX was acting very odd by ignoring the filetype case.